src/git-source-provider.ts

```diff
@@ -0,0 +1,285 @@
+import * as core from '@actions/core';
+import * as coreCommand from '@actions/core/lib/command';
+import * as fs from 'fs';
+import * as fsHelper from './fs-helper';
+import * as gitCommandManager from './git-command-manager';
+import * as io from '@actions/io';
+import * as path from 'path';
+
+let authConfigKey = `http.https://github.com/.extraheader`;
+
+export async function getSource(
+    repositoryPath: string,
+    repositoryOwner: string,
+    repositoryName: string,
+    ref: string,
+    commit: string,
+    clean: boolean,
+    fetchDepth: number,
+    lfs: boolean,
+    accessToken: string) {
+
+    core.info(`Syncing repository: ${repositoryOwner}/${repositoryName}`);
+    let repositoryUrl = `https://github.com/${encodeURIComponent(repositoryOwner)}/${encodeURIComponent(repositoryName)}`;
```

Comment:
should use const here if not redefining this later, explicitly define it as : string

Reply:
added prettier/eslint